core-bundle/src/Resources/contao/library/Contao/Widget.php

```diff
@@ -1243,6 +1243,29 @@ public static function getAttributesFromDca($arrData, $strName, $varValue=null,
 	{
 		$arrAttributes = $arrData['eval'] ?? array();
 
+		if (method_exists(System::getContainer(), 'getParameterBag'))
+		{
+			$objParameterBag = System::getContainer()->getParameterBag();
+
+			foreach ($arrAttributes as $strAttrKey => $varAttrValue)
+			{
+				if (!\is_string($varAttrValue) || !preg_match('/%[a-z][a-z0-9_]*\.[a-z0-9_.]+%/i', $varAttrValue, $arrMatches))
+				{
+					continue;
+				}
+
+				$varAttrValue = $objParameterBag->resolveValue($varAttrValue);
+				$varAttrValue = $objParameterBag->unescapeValue($varAttrValue);
+
+				if ($strAttrKey === 'extensions' && \is_array($varAttrValue))
```

Comment:
Any idea how we could abstract this, too? Maybe we can adjust the `__set()` method to handle both an array and a comma-separated list, so we do not have to convert the value here?

https://github.com/contao/contao/blob/39e3215c4886350fb65137cff9dc422dbe7de67b/core-bundle/src/Resources/contao/library/Contao/Widget.php#L218-L224

Reply:
How about this 22847d5d3d7875b766fd2120e869b10b5934ad59 ?